ci/repo_cleanups/remove_stale_rpm.sh

```diff
@@ -0,0 +1,231 @@
+
+        echo $(( ${doy2} - ${doy1} + ${days_to_add}))
+
+    }
+
+ 
+    ###################################################################################
+    # Get the list (has) of git branches that are still "live"
+    for key in `git_branch_info`; do
+       git_branches[$key]=1
+    done
+    
+    
+    origin=$PWD
+    cd ${rpm_repo_dir}
+    for dr in $(find ./* -maxdepth 0 -type d); do
+       dr=${dr##*\/}
+    
+       if [[ "$dr" = "." ]]; then
+           continue            ## Ignore the current directory
+       fi
+    
+       branch_name=${git_branches[$dr]}
+       if [[ "${branch_name}" = "master" ]]; then
+           continue
+       fi
+       if [[ "${branch_name}" = "develop" ]]; then
+           continue
+       fi
+   
+       ## If the branch name is *not* in the git_branches hash, remove it! 
+       if [[ -z ${branch_name} ]]; then
+            remove_dir ${dr}
+            if [ $? -ne 0 ]; then
+                exit 1
+            fi
+       fi
+    
+    done
+    
+    ## Now delete superfluous rpm's from the master directory
+    cd master
+    if [[ $? -ne 0 ]]; then
+        exit 0                # There's a problem here, but we won't worry about it 
+    fi
+   
+    nrepos=$(ls -1 . | wc -l)
+    if [[ ${nrepos} -lt 2 ]]; then
+        echo "Something is wrong. The master directory contains one or less repos. Quitting."
+        exit 1
+    fi
+ 
+    current=$(readlink current)
+    if [[ -z ${current} ]]; then
+        echo "No 'current' symlink in master! "
+        exit 1                # There is no "current" symlink. Don't remove anything!
+    fi
+    echo "'current' rpm repo is ${current}"
+
+    ## Just to be safe! We'll do the same 
+    readlink current
+    if [[ $? -ne 0 ]]; then
+        echo "No 'current' symlink in master! "
+        exit 1
+    fi
+
+    current=${current##*\/}
+    now=$(today)
+    echo "Checking for stale rpm repos under master..."
+    for dt in $(ls -d 2*); do
+        rpmdate=${dt:0:8}     # yyyymmdd is the format
+
+        ndays=$(delta_dates ${rpmdate} ${now})
+
+        if [[ "${dt}" = "${current}" ]]; then
+            continue
+        fi
+
+        if [[ ${ndays} -gt 14 ]]; then
+            ## Just in case the continue command does not work as I think it should...
+            if [[ "${dt}" = "${current}" ]]; then
+               echo "Will not delete 'current' (${current})"
+            else
+               echo "rm -rf  master/${dt} "
+               rm -rf  ./${dt}  
+               if [ $? -ne 0 ]; then
+                   echo "       rm FAILED for master/${dt} "
+               fi
+            fi
+        fi
+    done
+
+    cd ${origin}
+
+EOF
+)
+
+    echo "${var}"
+
+}
+
+#########################################################################################
+git_branches="$(run_git)"
+
+script=$(write_script)
+
+## Send the script generated above to the dh machine
+echo -e "${script/_STUB_/${git_branches}/}" | $SSH_REMOTE /usr/bin/bash 
```

Comment:
Writing the whole script in a function is not a good idea I think. For starters it removes syntax highlighting and it's likely to have other side effects too.

If it is required to send the script to another machine to execute it there, it's better to save the script normally as its own file and then send that through SSH either directly from the Jenkinsfile or from a wrapper script.